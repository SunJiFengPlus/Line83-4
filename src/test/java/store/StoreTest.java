package store;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class StoreTest {

    @Test
    void test() {
        CombinationApprovals.verifyAllCombinations(
                this::update,
                new String[]{"foo", "Aged Wine", "Show Ticket", "Sulfuras"},
                new Integer[]{-1, 0, 1, 10, 11, 12},
                new Integer[]{-1, 0, 1, 49, 50, 51}
        );
    }

    Item update(String name, int sellIn, int value) {
        Item[] items = new Item[]{new Item(name, sellIn, value)};
        Store store = new Store(items);
        store.updateValue();
        return items[0];
    }
}