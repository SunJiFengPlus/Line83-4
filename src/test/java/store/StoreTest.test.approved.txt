[foo, -1, 0] => foo, -2, 0 
[foo, -1, 1] => foo, -2, 0 
[foo, -1, 49] => foo, -2, 47 
[foo, -1, 50] => foo, -2, 48 
[foo, 0, 0] => foo, -1, 0 
[foo, 0, 1] => foo, -1, 0 
[foo, 0, 49] => foo, -1, 47 
[foo, 0, 50] => foo, -1, 48 
[foo, 1, 0] => foo, 0, 0 
[foo, 1, 1] => foo, 0, 0 
[foo, 1, 49] => foo, 0, 48 
[foo, 1, 50] => foo, 0, 49 
[foo, 10, 0] => foo, 9, 0 
[foo, 10, 1] => foo, 9, 0 
[foo, 10, 49] => foo, 9, 48 
[foo, 10, 50] => foo, 9, 49 
[foo, 11, 0] => foo, 10, 0 
[foo, 11, 1] => foo, 10, 0 
[foo, 11, 49] => foo, 10, 48 
[foo, 11, 50] => foo, 10, 49 
[foo, 12, 0] => foo, 11, 0 
[foo, 12, 1] => foo, 11, 0 
[foo, 12, 49] => foo, 11, 48 
[foo, 12, 50] => foo, 11, 49 
[Aged Wine, -1, 0] => Aged Wine, -2, 2 
[Aged Wine, -1, 1] => Aged Wine, -2, 3 
[Aged Wine, -1, 49] => Aged Wine, -2, 50 
[Aged Wine, -1, 50] => Aged Wine, -2, 50 
[Aged Wine, 0, 0] => Aged Wine, -1, 2 
[Aged Wine, 0, 1] => Aged Wine, -1, 3 
[Aged Wine, 0, 49] => Aged Wine, -1, 50 
[Aged Wine, 0, 50] => Aged Wine, -1, 50 
[Aged Wine, 1, 0] => Aged Wine, 0, 1 
[Aged Wine, 1, 1] => Aged Wine, 0, 2 
[Aged Wine, 1, 49] => Aged Wine, 0, 50 
[Aged Wine, 1, 50] => Aged Wine, 0, 50 
[Aged Wine, 10, 0] => Aged Wine, 9, 1 
[Aged Wine, 10, 1] => Aged Wine, 9, 2 
[Aged Wine, 10, 49] => Aged Wine, 9, 50 
[Aged Wine, 10, 50] => Aged Wine, 9, 50 
[Aged Wine, 11, 0] => Aged Wine, 10, 1 
[Aged Wine, 11, 1] => Aged Wine, 10, 2 
[Aged Wine, 11, 49] => Aged Wine, 10, 50 
[Aged Wine, 11, 50] => Aged Wine, 10, 50 
[Aged Wine, 12, 0] => Aged Wine, 11, 1 
[Aged Wine, 12, 1] => Aged Wine, 11, 2 
[Aged Wine, 12, 49] => Aged Wine, 11, 50 
[Aged Wine, 12, 50] => Aged Wine, 11, 50 
[Show Ticket, -1, 0] => Show Ticket, -2, 0 
[Show Ticket, -1, 1] => Show Ticket, -2, 0 
[Show Ticket, -1, 49] => Show Ticket, -2, 0 
[Show Ticket, -1, 50] => Show Ticket, -2, 0 
[Show Ticket, 0, 0] => Show Ticket, -1, 0 
[Show Ticket, 0, 1] => Show Ticket, -1, 0 
[Show Ticket, 0, 49] => Show Ticket, -1, 0 
[Show Ticket, 0, 50] => Show Ticket, -1, 0 
[Show Ticket, 1, 0] => Show Ticket, 0, 3 
[Show Ticket, 1, 1] => Show Ticket, 0, 4 
[Show Ticket, 1, 49] => Show Ticket, 0, 50 
[Show Ticket, 1, 50] => Show Ticket, 0, 50 
[Show Ticket, 10, 0] => Show Ticket, 9, 2 
[Show Ticket, 10, 1] => Show Ticket, 9, 3 
[Show Ticket, 10, 49] => Show Ticket, 9, 50 
[Show Ticket, 10, 50] => Show Ticket, 9, 50 
[Show Ticket, 11, 0] => Show Ticket, 10, 1 
[Show Ticket, 11, 1] => Show Ticket, 10, 2 
[Show Ticket, 11, 49] => Show Ticket, 10, 50 
[Show Ticket, 11, 50] => Show Ticket, 10, 50 
[Show Ticket, 12, 0] => Show Ticket, 11, 1 
[Show Ticket, 12, 1] => Show Ticket, 11, 2 
[Show Ticket, 12, 49] => Show Ticket, 11, 50 
[Show Ticket, 12, 50] => Show Ticket, 11, 50 
[Sulfuras, -1, 0] => Sulfuras, -1, 0 
[Sulfuras, -1, 1] => Sulfuras, -1, 1 
[Sulfuras, -1, 49] => Sulfuras, -1, 49 
[Sulfuras, -1, 50] => Sulfuras, -1, 50 
[Sulfuras, 0, 0] => Sulfuras, 0, 0 
[Sulfuras, 0, 1] => Sulfuras, 0, 1 
[Sulfuras, 0, 49] => Sulfuras, 0, 49 
[Sulfuras, 0, 50] => Sulfuras, 0, 50 
[Sulfuras, 1, 0] => Sulfuras, 1, 0 
[Sulfuras, 1, 1] => Sulfuras, 1, 1 
[Sulfuras, 1, 49] => Sulfuras, 1, 49 
[Sulfuras, 1, 50] => Sulfuras, 1, 50 
[Sulfuras, 10, 0] => Sulfuras, 10, 0 
[Sulfuras, 10, 1] => Sulfuras, 10, 1 
[Sulfuras, 10, 49] => Sulfuras, 10, 49 
[Sulfuras, 10, 50] => Sulfuras, 10, 50 
[Sulfuras, 11, 0] => Sulfuras, 11, 0 
[Sulfuras, 11, 1] => Sulfuras, 11, 1 
[Sulfuras, 11, 49] => Sulfuras, 11, 49 
[Sulfuras, 11, 50] => Sulfuras, 11, 50 
[Sulfuras, 12, 0] => Sulfuras, 12, 0 
[Sulfuras, 12, 1] => Sulfuras, 12, 1 
[Sulfuras, 12, 49] => Sulfuras, 12, 49 
[Sulfuras, 12, 50] => Sulfuras, 12, 50 
[Cure, -1, 0] => Cure, -2, 0 
[Cure, -1, 1] => Cure, -2, 0 
[Cure, -1, 49] => Cure, -2, 45 
[Cure, -1, 50] => Cure, -2, 46 
[Cure, 0, 0] => Cure, -1, 0 
[Cure, 0, 1] => Cure, -1, 0 
[Cure, 0, 49] => Cure, -1, 45 
[Cure, 0, 50] => Cure, -1, 46 
[Cure, 1, 0] => Cure, 0, 0 
[Cure, 1, 1] => Cure, 0, 0 
[Cure, 1, 49] => Cure, 0, 47 
[Cure, 1, 50] => Cure, 0, 48 
[Cure, 10, 0] => Cure, 9, 0 
[Cure, 10, 1] => Cure, 9, 0 
[Cure, 10, 49] => Cure, 9, 47 
[Cure, 10, 50] => Cure, 9, 48 
[Cure, 11, 0] => Cure, 10, 0 
[Cure, 11, 1] => Cure, 10, 0 
[Cure, 11, 49] => Cure, 10, 47 
[Cure, 11, 50] => Cure, 10, 48 
[Cure, 12, 0] => Cure, 11, 0 
[Cure, 12, 1] => Cure, 11, 0 
[Cure, 12, 49] => Cure, 11, 47 
[Cure, 12, 50] => Cure, 11, 48 
